GitUI/UserControls/RevisionGrid.cs

```diff
@@ -1687,18 +1693,18 @@ private void RevisionsCellPainting(object sender, DataGridViewCellPaintingEventA
                         int gravatarTop = e.CellBounds.Top + textHeight + 6;
                         int gravatarLeft = e.CellBounds.Left + baseOffset + 2;
 
-
-                        Image gravatar = Gravatar.GravatarService.GetImageFromCache(revision.AuthorEmail + gravatarSize.ToString() + ".png", revision.AuthorEmail, AppSettings.AuthorImageCacheDays, gravatarSize, AppSettings.GravatarCachePath, FallBackService.MonsterId);
-
-                        if (gravatar == null && !string.IsNullOrEmpty(revision.AuthorEmail))
+                        var gravatar = _avatarCache.GetImage($"{revision.AuthorEmail}.png", null);
+                        if (gravatar == null)
                         {
-                            ThreadPool.QueueUserWorkItem(o =>
-                            Gravatar.GravatarService.LoadCachedImage(revision.AuthorEmail + gravatarSize.ToString() + ".png", revision.AuthorEmail, null, AppSettings.AuthorImageCacheDays, gravatarSize, AppSettings.GravatarCachePath, RefreshGravatar, FallBackService.MonsterId));
+                            gravatar = Resources.User;
+                            // kick off download operation, will likely display the avatar during the next round of repaint
+                            await _gravatarService.GetAvatarAsync(revision.AuthorEmail, AppSettings.AuthorImageSize, AppSettings.GravatarDefaultImageType)
+                                .ContinueWith(t => this.InvokeAsync(() =>
+                                {
+                                    Revisions.InvalidateCell(e.ColumnIndex, e.RowIndex);
```

Comment:
Yes, I think you're right... I shouldn't cut any code before breakfast.
I'll test it over again